pkg/registry/registry_to_model.go

```diff
@@ -0,0 +1,235 @@
+		Description: desc,
+		Icon:        mIcon,
+		Channels:    make(map[string]*model.Channel),
+	}
+
+	mb, err := registryBundleToModelBundle(b)
+	mb.Package = pkg
+	if err != nil {
+		return nil, err
+	}
+
+	for _, ch := range extractChannels(b.Annotations.Channels) {
+		newCh := &model.Channel{
+			Name: ch,
+		}
+		chBundle := mb
+		chBundle.Channel = newCh
+		bundles = append(bundles, *chBundle)
+	}
+	return bundles, nil
+}
+
+func registryBundleToModelBundle(b *Bundle) (*model.Bundle, error) {
+	bundleProps, err := convertRegistryBundleToModelProperties(b)
+	if err != nil {
+		return nil, fmt.Errorf("error converting properties for internal model: %v", err)
+	}
+
+	csv, err := b.ClusterServiceVersion()
+	if err != nil {
+		return nil, fmt.Errorf("Could not get CVS for bundle: %s", err)
+	}
+	replaces, err := csv.GetReplaces()
+	if err != nil {
+		return nil, fmt.Errorf("Could not get Replaces from CSV for bundle: %s", err)
+	}
+	skips, err := csv.GetSkips()
+	if err != nil {
+		return nil, fmt.Errorf("Could not get Skips from CSV for bundle: %s", err)
+	}
+	relatedImages, err := convertToModelRelatedImages(csv)
+	if err != nil {
+		return nil, fmt.Errorf("Could not get Related images from bundle: %v", err)
+	}
+
+	return &model.Bundle{
+		Name:          csv.Name,
+		Image:         b.BundleImage,
+		Replaces:      replaces,
+		Skips:         skips,
+		Properties:    bundleProps,
+		RelatedImages: relatedImages,
+	}, nil
+}
+
+func convertRegistryBundleToModelProperties(b *Bundle) ([]property.Property, error) {
+	var out []property.Property
+
+	skips, err := b.csv.GetSkips()
+	if err != nil {
+		return nil, fmt.Errorf("Could not get Skips from CSV for bundle: %s", err)
+	}
+
+	for _, skip := range skips {
+		out = append(out, property.MustBuildSkips(skip))
+	}
+
+	skipRange := b.csv.GetSkipRange()
+	if skipRange != "" {
+		out = append(out, property.MustBuildSkipRange(skipRange))
+	}
+
+	replaces, err := b.csv.GetReplaces()
+	if err != nil {
+		return nil, fmt.Errorf("Could not get Replaces from CSV for bundle: %s", err)
+	}
+	for _, ch := range extractChannels(b.Annotations.Channels) {
+		out = append(out, property.MustBuildChannel(ch, replaces))
+	}
+
+	providedGVKs := map[property.GVK]struct{}{}
+	requiredGVKs := map[property.GVKRequired]struct{}{}
+
+	foundPackageProperty := false
+	for i, p := range b.Properties {
+		switch p.Type {
+		case property.TypeGVK:
+			var v property.GVK
+			if err := json.Unmarshal(json.RawMessage(p.Value), &v); err != nil {
+				return nil, property.ParseError{Idx: i, Typ: p.Type, Err: err}
+			}
+			k := property.GVK{Group: v.Group, Kind: v.Kind, Version: v.Version}
+			providedGVKs[k] = struct{}{}
+		case property.TypePackage:
+			var v property.Package
+			if err := json.Unmarshal(json.RawMessage(p.Value), &v); err != nil {
+				return nil, property.ParseError{Idx: i, Typ: p.Type, Err: err}
+			}
+			out = append(out, property.MustBuildPackageRequired(v.PackageName, v.Version))
```

Comment:
This seems incorrect. We should just be converting the `olm.package` properties directly (not changing them to `PackageRequired` properties). Also, shouldn't we be setting `foundPackageProperty = true` here?

This may have been a screw up on my part since, since I know you cribbed from me.